File: src/flink-examples/flink-examples-security/FlinkPipelineScalaExample/src/main/java/com/huawei/bigdata/flink/examples/TestPipeline_NettySource2.scala
package com.huawei.bigdata.flink.examples

import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment
import org.apache.flink.streaming.connectors.netty.source.NettySource
import org.apache.flink.streaming.connectors.netty.utils.ZookeeperRegisterServerHandler
import org.apache.flink.streaming.api.scala._

import scala.util.Random


object TestPipeline_NettySource2 {

  def main(args: Array[String]): Unit = {

    val env = StreamExecutionEnvironment.getExecutionEnvironment
    env.setParallelism(2)

    val zkRegisterServerHandler = new ZookeeperRegisterServerHandler
    env.addSource(new NettySource("NettySource-2", "TOPIC-2", zkRegisterServerHandler))
      .map(x=>(2, new String(x)))
      .filter(x=>{
        Random.nextInt(50000) == 10
      })
      .print()

    env.execute()
  }
}

File: src/flink-examples/flink-examples-normal/FlinkPipelineJavaExample/src/main/java/com/huawei/bigdata/flink/examples/TestPipelineNettySource2.java
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2012-2020. All rights reserved.
 */

package com.huawei.bigdata.flink.examples;

import org.apache.flink.api.common.functions.MapFunction;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.connectors.netty.source.NettySource;
import org.apache.flink.streaming.connectors.netty.utils.ZookeeperRegisterServerHandler;

import java.nio.charset.Charset;

/**
 * NettySource excmple 2
 * 
 * @since 2019/9/30
 */
public class TestPipelineNettySource2 {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(2);

        ZookeeperRegisterServerHandler zkRegisterServerHandler = new ZookeeperRegisterServerHandler();
        env.addSource(new NettySource("NettySource-2", "TOPIC-2", zkRegisterServerHandler))
                .map(
                        new MapFunction<byte[], String>() {
                            @Override
                            public String map(byte[] bytes) {
                                return new String(bytes, Charset.forName("UTF-8"));
                            }
                        })
                .print();

        env.execute();
    }
}
